DevoxxClientMobile/src/main/java/com/devoxx/service/DevoxxService.java

```diff
@@ -746,10 +697,25 @@ private void retrieveExhibitionMapsInternal() {
 
 	@Override
 	public GluonObservableObject<Sponsor> retrieveSponsorByCode(String sponsorCode) {
-		RemoteFunctionObject fnSponsorByCode = RemoteFunctionBuilder.create("sponsor")
-				.param("conferenceId", getConference().getId()).param("sponsorCode", sponsorCode).object();
-		GluonObservableObject<Sponsor> sponsor = fnSponsorByCode.call(Sponsor.class);
-		sponsor.setOnFailed(e -> LOG.log(Level.WARNING, String.format(REMOTE_FUNCTION_FAILED_MSG, "sponsor"),
+		final GluonObservableObject<Sponsor> sponsor = new GluonObservableObject<>();
+
+		final GluonObservableList<Sponsor> sponsors = retrieveSponsors();
+		sponsors.setOnSucceeded(e -> {
+			sponsors.stream()
+					.filter(s -> s.getCode().equals(sponsorCode))
+					.findFirst()
+					.map(s -> {
```

Comment:
you should try to use peek to modify objects in a stream

Reply:
I don't want to modify the object in the stream. I want to get an optional sponsor, so if the sponsor exist I do one thing, but if it doesn't exist I do other stuff.

Reply:
I don't want to modify the object in the stream. I want to get an optional sponsor, so if the sponsor exist I do one thing, but if it doesn't exist I do other stuff.

Reply:
I see now, but then, the use of map here is missleading because there's no modification on the maped object `s`, you actually modifing a third object our of the scope of the stream.
Wouldn't it be easier to read just doing just what you explained in words:
1. Obtain the optional from the stream
2. if (optional.isPresent()) { update s with success } else { do other stuff}

Reply:
Done, thanks @abelsromero  for your review